### ISSUE TITLE
Notes are duplicated for some split transactions

### ISSUE THREAD

**Issue by gemellon** [moneymanagerex/moneymanagerex#6462]
MMEX version:
 - [x] 1.6.4
 - [x] 1.7.0
 - [ ] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [x] Mac OSX
 - [ ] Linux 

# Description of the bug

If a user creates a new transaction that has a split, and enters a comment in the Notes field, under certain circumstances this comment is duplicated when viewing the transaction in the bottom pane.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

The order of the steps is important.

1. Create a new transaction
2. Enter the amount, choose payee and category - this is the first split
3. Enter a comment in the Notes field
4. Create a split with the '+' button - this is the second split
5. Enter an amount and category
6. Do not enter a note for this split
7. Click 'OK' and then 'OK' to finish the new transaction

Expected result:

If a user clicks on the new transaction the information will show in the bottom pane with the comment from the Notes field at the bottom just as it was entered by the user.

Actual result:

The comment in the Notes field is shown twice

![1](https://github.com/moneymanagerex/moneymanagerex/assets/145873120/01cf2099-398c-4fe1-9aef-c34b43daef69)

# Additional information

The steps above are the only way I can reproduce this problem. If I do something different such as enter a note for the split at point 6. or if I enter the note for the first split through the Split dialogue box instead of in the main New Transaction dialogue then this bug does not happen.

But the steps I show above are the step I make when I have a split transaction, because it is faster than making a separate note for each split in the Split dialogue. If I train myself I could work around this bug but perhaps it's better to fix it :)

**Comment by whalley** [comment]
I can reproduce the comment against the transaction is sometimes copied to the first split.

**Comment by n-stein** [comment]
What is the expected behavior here? 

Based on your workflow it seems you're requesting that a note entered in the transaction dialog before clicking the split button be transferred to the first split the same way the category is transferred, but I'm not sure I agree. It makes sense for the category to transfer because that field is exclusive -- a category is either on the transaction or the split, but never both. However, this is not the case with notes since you can assign different notes to the parent transaction and the splits.

**Comment by whalley** [comment]
THinking about this... My view on this would be to keep transaction notes seperate from the individual split notes. Therefore not copy when splitting or unsplitting. But that is just how I tend to organise things.

**Comment by luisdev** [comment]
I agree, the split notes should be separate to the parent notes. You might want to have a note in the parent note field and other notes in each of the split note fields.

**Comment by n-stein** [comment]
Right now we copy Amount, Category, Tags, and Notes from the transaction to the first split only if no splits already exist AND the user has entered a valid Category into the category field first. If they click the split button without prefilling the category, we don't copy anything even the amount. Do we want to change this behavior?

Perhaps we should always copy the fields that are defined exclusively by the split dialog, and don't copy the ones that may be defined independently on both the transaction and splits? Suggestion would be to:

- Always copy amount (Amount becomes a calculated field on the parent once splits are created)
- Copy category to the first split (Category on the parent is disabled once splits are created)
- Don't copy tags or notes (since the user can specify these fields independently on parent transaction and splits)

**Comment by rmelillo76** [comment]
Hi @whalley I just tested #7754 and if only 1 split is entered, the notes and tags from that split are lost after clicking OK to exit the Split Transaction dialog and are not saved. They only seem to be retained when 2 or more splits are entered.

**Comment by whalley** [comment]
> Hi [@whalley](https://github.com/whalley) I just tested [#7754](https://github.com/moneymanagerex/moneymanagerex/pull/7754) and if only 1 split is entered, the notes and tags from that split are lost after clicking OK to exit the Split Transaction dialog and are not saved. They only seem to be retained when 2 or more splits are entered.

OK, this was intentional. As the user may have entered tag and note already in main transaction. We could get clever and check for this and concatenate or offer user an option to choose. Admittedly if tag/note not added in main transaction we could just copy but it is inconsistent.

Or are you referring to the case where the transaction is not yet posted, and the user exits the split transaction dialog and then goes back into it. I can see an anomaly here.

**Comment by rmelillo76** [comment]
> > Hi [@whalley](https://github.com/whalley) I just tested [#7754](https://github.com/moneymanagerex/moneymanagerex/pull/7754) and if only 1 split is entered, the notes and tags from that split are lost after clicking OK to exit the Split Transaction dialog and are not saved. They only seem to be retained when 2 or more splits are entered.
> 
> OK, this was intentional. As the user may have entered tag and note already in main transaction. We could get clever and check for this and concatenate or offer user an option to choose. Admittedly if tag/note not added in main transaction we could just copy but it is inconsistent.
> 
> Or are you referring to the case where the transaction is not yet posted, and the user exits the split transaction dialog and then goes back into it. I can see an anomaly here.

It just seems confusing now that a user could end up selecting tags and entering notes which are not saved, both when exiting the dialog as well as upon returning to the parent transaction and saving. Just doesn't seem that intuitive, but that could just be me. I thought the old way of transferring the info from the main transaction to the first split was OK, and the byproduct of 'duplicates' was a logical outcome. But again, just a personal preference.
